components/mdi/Core/Src/main.c

```diff
@@ -211,7 +215,21 @@ int main(void)
     //////////////////////////////////////
     // SEND SIGNALS TO MOTOR CONTROLLER //
     //////////////////////////////////////
-    if( (send_data_flag == 1 ) && (Parse_Data_Flag == 0) )
+
+    // Check for mcb timeout
+    if(HAL_GetTick() > msg0.last_rx_time + MCB_COMMUNICATION_TIMEOUT)
+    {
+      if(0 == mcb_communication_fault)
+      {
+        // Send 0.0f acceleration
+        Send_Voltage(0.0f, DAC_ACC_ADDR, &hi2c2);
+        // Send 0.0f regen
+        Send_Voltage(0.0f, DAC_REGEN_ADDR, &hi2c2);
```

Comment:
Checked this in an onine compiler. Floating point 0 * 1024 typecasted to uint16_t returns 0.